mongo/integration/mtest/mongotest.go

```diff
@@ -666,13 +666,26 @@ func verifyTopologyConstraints(topologies []TopologyKind) error {
 	return fmt.Errorf("topology kind %q does not match any of the required kinds %q", testContext.topoKind, topologies)
 }
 
+func verifyServerParametersConstraints(serverParameters map[string]bool) error {
+	for param, expectedValue := range serverParameters {
+		actualValue, ok := testContext.serverParameters.Lookup(param).BooleanOK()
```

Comment:
I was going to ask what happens when a future test adds a non-bool values for `serverParameters`. If that happens, IIUC we'll error here. So it should be clear from then that we need to update the test runner to accommodate.

Reply:
Technically, I think that there would be an error during unmarshalling of the `RunOnBlock` (creating the `map[string]bool`) before it got to this point. If non-bool values are introduced to `serverParameters` though, I guess we'd want to store it as raw BSON and loop through it. I could switch to doing that now; will discuss offline.

Reply:
Ok, if it is a straightforward change, I suggest changing it now. (I believe `bson.D` or `bsoncore.BuildDocument` were mentioned offline).

Reply:
`bson.D` and the `bsoncore` builders were suggested for GODRIVER-1805. For this, I suggested changing to `map[string]bson.RawValue` and comparing the values directly.

Reply:
Cool, done!